### ISSUE TITLE
Stuck Processing repo map

### ISSUE THREAD

**Issue by ShawnSack** [Aider-AI/aider#2394]
### Issue

Aider gets stuck on /map. My repo only has 424 files, but when I attempt to talk with LLM or run /map it gets stuck processing. Tested with multiple model with same behavior. Uninstalled and re-installed. Cleared history, etc. No dirty commits. Anything else I could check for? Unable to use Aider because of this. 

```
Config files search order, if no --config:
  - /workspaces/nextgen/.aider.conf.yml 
  - /home/node/.aider.conf.yml 
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Loaded /workspaces/nextgen/.env.local
Too soon to check version: 2.3 hours
Command Line Args:   --config .aider.conf.yaml --verbose

Environment Variables:
  OPENAI_API_KEY:    ...G27p
Config File (.aider.conf.yaml):
  model:             bedrock/anthropic.claude-3-5-sonnet-20241022-v2:0
  4o:                False
  weak-model:        gpt-4o-mini
  env-file:          .env.local
  cache-prompts:     True
  map-tokens:        2048
  lint-cmd:          ['typescript: pnpm run lint --file']
  auto-lint:         True
  read:              ['conventions.md', 'package.json', 'tsconfig.json', '.eslintrc.js', 'src/docs/components/']

Defaults:
  --model-settings-file:.aider.model.settings.yml
  --model-metadata-file:.aider.model.metadata.json
  --cache-keepalive-pings:0
  --map-refresh:     auto
  --map-multiplier-no-files:2
  --input-history-file:/workspaces/nextgen/.aider.input.history
  --chat-history-file:/workspaces/nextgen/.aider.chat.history.md
  --user-input-color:#00cc00
  --tool-error-color:#FF2222
  --tool-warning-color:#FFA500
  --assistant-output-color:#0088ff
  --code-theme:      default
  --aiderignore:     /workspaces/nextgen/.aiderignore
  --test-cmd:        []
  --encoding:        utf-8
  --voice-format:    wav
  --voice-language:  en

Option settings:
  - aiderignore: /workspaces/nextgen/.aiderignore
  - analytics: False
  - analytics_disable: False
  - analytics_log: None
  - anthropic_api_key: None
  - apply: None
  - apply_clipboard_edits: False
  - assistant_output_color: #0088ff
  - attribute_author: True
  - attribute_commit_message_author: False
  - attribute_commit_message_committer: False
  - attribute_committer: True
  - auto_commits: True
  - auto_lint: True
  - auto_test: False
  - cache_keepalive_pings: 0
  - cache_prompts: True
  - chat_history_file: /workspaces/nextgen/.aider.chat.history.md
  - chat_language: None
  - check_update: True
  - code_theme: default
  - commit: False
  - commit_prompt: None
  - completion_menu_bg_color: None
  - completion_menu_color: None
  - completion_menu_current_bg_color: None
  - completion_menu_current_color: None
  - config: .aider.conf.yaml
  - dark_mode: False
  - dirty_commits: True
  - dry_run: False
  - edit_format: None
  - editor_edit_format: None
  - editor_model: None
  - encoding: utf-8
  - env_file: .env.local
  - exit: False
  - fancy_input: True
  - file: None
  - files: []
  - git: True
  - gitignore: True
  - gui: False
  - input_history_file: /workspaces/nextgen/.aider.input.history
  - install_main_branch: False
  - just_check_update: False
  - light_mode: False
  - lint: False
  - lint_cmd: ['typescript: pnpm run lint --file']
  - list_models: None
  - llm_history_file: None
  - load: None
  - map_multiplier_no_files: 2
  - map_refresh: auto
  - map_tokens: 2048
  - max_chat_history_tokens: None
  - message: None
  - message_file: None
  - model: bedrock/anthropic.claude-3-5-sonnet-20241022-v2:0
  - model_metadata_file: .aider.model.metadata.json
  - model_settings_file: .aider.model.settings.yml
  - openai_api_base: None
  - openai_api_deployment_id: None
  - openai_api_key: ...G27p
  - openai_api_type: None
  - openai_api_version: None
  - openai_organization_id: None
  - pretty: True
  - read: ['conventions.md', 'package.json', 'tsconfig.json', '.eslintrc.js', 'src/docs/components/']
  - restore_chat_history: False
  - show_diffs: False
  - show_model_warnings: True
  - show_prompts: False
  - show_repo_map: False
  - skip_sanity_check_repo: False
  - stream: True
  - subtree_only: False
  - suggest_shell_commands: True
  - test: False
  - test_cmd: []
  - tool_error_color: #FF2222
  - tool_output_color: None
  - tool_warning_color: #FFA500
  - upgrade: False
  - user_input_color: #00cc00
  - verbose: True
  - verify_ssl: True
  - vim: False
  - voice_format: wav
  - voice_language: en
  - weak_model: gpt-4o-mini
  - yes_always: None

Checking imports for version 0.63.2 and executable /usr/local/py-utils/venvs/aider-chat/bin/python
Installs file: /home/node/.aider/installs.json
Installs file exists and loaded
Not first run, loading imports in background thread
No model settings files loaded
Searched for model settings files:
  - /home/node/.aider.model.settings.yml
  - /workspaces/nextgen/.aider.model.settings.yml
Loaded model metadata from:
  - /usr/local/py-utils/venvs/aider-chat/lib/python3.11/site-packages/aider/resources/model-metadata.json
Model info:
{
    "key": "anthropic.claude-3-5-sonnet-20241022-v2:0",
    "max_tokens": 8192,
    "max_input_tokens": 200000,
    "max_output_tokens": 8192,
    "input_cost_per_token": 3e-06,
    "cache_creation_input_token_cost": null,
    "cache_read_input_token_cost": null,
    "input_cost_per_character": null,
    "input_cost_per_token_above_128k_tokens": null,
    "input_cost_per_query": null,
    "input_cost_per_second": null,
    "input_cost_per_audio_token": null,
    "output_cost_per_token": 1.5e-05,
    "output_cost_per_audio_token": null,
    "output_cost_per_character": null,
    "output_cost_per_token_above_128k_tokens": null,
    "output_cost_per_character_above_128k_tokens": null,
    "output_cost_per_second": null,
    "output_cost_per_image": null,
    "output_vector_size": null,
    "litellm_provider": "bedrock",
    "mode": "chat",
    "supported_openai_params": [
        "max_tokens",
        "max_completion_tokens",
        "stream",
        "stream_options",
        "stop",
        "temperature",
        "top_p",
        "extra_headers",
        "response_format"
    ],
    "supports_system_messages": null,
    "supports_response_schema": null,
    "supports_vision": true,
    "supports_function_calling": true,
    "supports_assistant_prefill": true,
    "supports_prompt_caching": false,
    "supports_audio_input": false,
    "supports_audio_output": false
}
RepoMap initialized with map_mul_no_files: 2
Aider v0.63.2
Main model: bedrock/anthropic.claude-3-5-sonnet-20241022-v2:0 with diff edit format, prompt cache, infinite output
Weak model: gpt-4o-mini
Git repo: .git with 424 files
Repo-map: using 2048 tokens, files refresh
VSCode terminal detected, pretty output has been disabled.
Use /help <question> for help, run "aider --help" to see cmd line args
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Readonly: .eslintrc.js conventions.md package.json src/docs/components/form-fields.md tsconfig.json                                    
> /map                                                                                                                                 

Processing /workspaces/nextgen/.devcontainer/devcontainer.json
Processing /workspaces/nextgen/.dockerignore
Processing /workspaces/nextgen/.editorconfig
Processing /workspaces/nextgen/.env.example
Processing /workspaces/nextgen/.eslintrc.js
Processing /workspaces/nextgen/.gitattributes
Processing /workspaces/nextgen/.gitignore
Processing /workspaces/nextgen/.prettierignore
Processing /workspaces/nextgen/.vscode/settings.json
Processing /workspaces/nextgen/Dockerfile
Processing /workspaces/nextgen/README.md
Processing /workspaces/nextgen/apollo.config.json
Processing /workspaces/nextgen/codegen.ts
Processing /workspaces/nextgen/conventions.md
Processing /workspaces/nextgen/graphql.schema.json
Processing /workspaces/nextgen/image.png
Processing /workspaces/nextgen/jest.config.mjs
Processing /workspaces/nextgen/next.config.mjs
Processing /workspaces/nextgen/package.json
Processing /workspaces/nextgen/pnpm-lock.yaml
Processing /workspaces/nextgen/prettier.config.mjs
Processing /workspaces/nextgen/src/app/auth/custom/layout.tsx
Processing /workspaces/nextgen/src/app/auth/custom/reset-password/page.tsx
Processing /workspaces/nextgen/src/app/auth/custom/sign-in/page.tsx
Processing /workspaces/nextgen/src/app/auth/custom/sign-up/page.tsx
Processing /workspaces/nextgen/src/app/dashboard/layout.tsx
Processing /workspaces/nextgen/src/app/dashboard/page.tsx
Processing /workspaces/nextgen/src/app/dashboard/reports/REP001/page.generated.tsx
Processing /workspaces/nextgen/src/app/dashboard/reports/REP001/page.tsx
Processing /workspaces/nextgen/src/app/dashboard/reports/page.tsx
Processing /workspaces/nextgen/src/app/dashboard/subscriptions/mutations.generated.tsx
Processing /workspaces/nextgen/src/app/dashboard/subscriptions/mutations.ts
Processing /workspaces/nextgen/src/app/dashboard/subscriptions/page.tsx
Processing /workspaces/nextgen/src/app/dashboard/subscriptions/query.generated.tsx
Processing /workspaces/nextgen/src/app/dashboard/subscriptions/query.ts
Processing /workspaces/nextgen/src/app/layout.tsx
Processing /workspaces/nextgen/src/app/marketing/layout.tsx
Processing /workspaces/nextgen/src/app/marketing/page.tsx
Processing /workspaces/nextgen/src/app/not-found.tsx
Processing /workspaces/nextgen/src/app/page.tsx
Processing /workspaces/nextgen/src/components/auth/centered-layout.tsx
Processing /workspaces/nextgen/src/components/auth/custom/reset-password-form.tsx
Processing /workspaces/nextgen/src/components/auth/custom/sign-in-form.tsx
Processing /workspaces/nextgen/src/components/auth/custom/sign-up-form.tsx
Processing /workspaces/nextgen/src/components/auth/guest-guard.tsx
Processing /workspaces/nextgen/src/components/auth/split-layout.tsx
Processing /workspaces/nextgen/src/components/auth/strategy-guard.tsx
Processing /workspaces/nextgen/src/components/core/analytics.tsx
Processing /workspaces/nextgen/src/components/core/apollo-provider.tsx
Processing /workspaces/nextgen/src/components/core/breadcrumbs-separator.tsx
Processing /workspaces/nextgen/src/components/core/code-highlighter.tsx
Processing /workspaces/nextgen/src/components/core/data-table.tsx
Processing /workspaces/nextgen/src/components/core/drawers/sub-drawer-tab.tsx
Processing /workspaces/nextgen/src/components/core/drawers/sub-drawer.tsx
Processing /workspaces/nextgen/src/components/core/dropdown/dropdown-context.tsx
Processing /workspaces/nextgen/src/components/core/dropdown/dropdown-popover.tsx
Processing /workspaces/nextgen/src/components/core/dropdown/dropdown-trigger.tsx
Processing /workspaces/nextgen/src/components/core/dropdown/dropdown.tsx
Processing /workspaces/nextgen/src/components/core/file-dropzone.tsx
Processing /workspaces/nextgen/src/components/core/file-icon.tsx
Processing /workspaces/nextgen/src/components/core/filter-button.tsx
Processing /workspaces/nextgen/src/components/core/form/__tests__/autocomplete-field.test.tsx
Processing /workspaces/nextgen/src/components/core/form/__tests__/date-field.test.tsx
Processing /workspaces/nextgen/src/components/core/form/__tests__/form-section.test.tsx
Processing /workspaces/nextgen/src/components/core/form/__tests__/number-field.test.tsx
Processing /workspaces/nextgen/src/components/core/form/__tests__/select-field.test.tsx
Processing /workspaces/nextgen/src/components/core/form/__tests__/text-field.test.tsx
Processing /workspaces/nextgen/src/components/core/form/autocomplete-field.tsx
Processing /workspaces/nextgen/src/components/core/form/date-field.tsx
Processing /workspaces/nextgen/src/components/core/form/form-field.tsx
Processing /workspaces/nextgen/src/components/core/form/form-section.tsx
Processing /workspaces/nextgen/src/components/core/form/index.ts
Processing /workspaces/nextgen/src/components/core/form/number-field.tsx
Processing /workspaces/nextgen/src/components/core/form/select-field.tsx
Processing /workspaces/nextgen/src/components/core/form/text-field.tsx
Processing /workspaces/nextgen/src/components/core/i18n-provider.tsx
Processing /workspaces/nextgen/src/components/core/localization-provider.tsx
Processing /workspaces/nextgen/src/components/core/logo.tsx
Processing /workspaces/nextgen/src/components/core/multi-select.tsx
Processing /workspaces/nextgen/src/components/core/no-ssr.tsx
Processing /workspaces/nextgen/src/components/core/option.tsx
Processing /workspaces/nextgen/src/components/core/pdf-viewer.tsx
Processing /workspaces/nextgen/src/components/core/presence.tsx
Processing /workspaces/nextgen/src/components/core/property-item.tsx
Processing /workspaces/nextgen/src/components/core/property-list.tsx
Processing /workspaces/nextgen/src/components/core/settings/option.tsx
Processing /workspaces/nextgen/src/components/core/settings/options-color-scheme.tsx
Processing /workspaces/nextgen/src/components/core/settings/options-direction.tsx
Processing /workspaces/nextgen/src/components/core/settings/options-layout.tsx
Processing /workspaces/nextgen/src/components/core/settings/options-nav-color.tsx
Processing /workspaces/nextgen/src/components/core/settings/options-primary-color.tsx
Processing /workspaces/nextgen/src/components/core/settings/settings-button.tsx
Processing /workspaces/nextgen/src/components/core/settings/settings-drawer.tsx
Processing /workspaces/nextgen/src/components/core/text-editor/text-editor-toolbar.tsx
Processing /workspaces/nextgen/src/components/core/text-editor/text-editor.tsx
Processing /workspaces/nextgen/src/components/core/theme-provider/emotion-cache.tsx
Processing /workspaces/nextgen/src/components/core/theme-provider/rtl.tsx
Processing /workspaces/nextgen/src/components/core/theme-provider/theme-provider.tsx
Processing /workspaces/nextgen/src/components/core/tip.tsx
Processing /workspaces/nextgen/src/components/core/toaster.tsx
Processing /workspaces/nextgen/src/components/dashboard/layout/config.ts
Processing /workspaces/nextgen/src/components/dashboard/layout/contacts-popover.tsx
Processing /workspaces/nextgen/src/components/dashboard/layout/dynamic-layout.tsx
Processing /workspaces/nextgen/src/components/dashboard/layout/horizontal/horizontal-layout.tsx
Processing /workspaces/nextgen/src/components/dashboard/layout/horizontal/main-nav.tsx
Processing /workspaces/nextgen/src/components/dashboard/layout/horizontal/styles.ts
Processing /workspaces/nextgen/src/components/dashboard/layout/language-popover.tsx
Processing /workspaces/nextgen/src/components/dashboard/layout/mobile-nav.tsx
Processing /workspaces/nextgen/src/components/dashboard/layout/nav-icons.tsx
Processing /workspaces/nextgen/src/components/dashboard/layout/notifications-popover.tsx
Processing /workspaces/nextgen/src/components/dashboard/layout/search-dialog.tsx
Processing /workspaces/nextgen/src/components/dashboard/layout/vertical/main-nav.tsx
Processing /workspaces/nextgen/src/components/dashboard/layout/vertical/side-nav.tsx
Processing /workspaces/nextgen/src/components/dashboard/layout/vertical/styles.ts
Processing /workspaces/nextgen/src/components/dashboard/layout/vertical/vertical-layout.tsx
Processing /workspaces/nextgen/src/components/dashboard/layout/workspaces-popover.tsx
Processing /workspaces/nextgen/src/components/dashboard/layout/workspaces-switch.tsx
Processing /workspaces/nextgen/src/components/dashboard/overview/app-chat.tsx
Processing /workspaces/nextgen/src/components/dashboard/overview/app-limits.tsx
Processing /workspaces/nextgen/src/components/dashboard/overview/app-usage.tsx
Processing /workspaces/nextgen/src/components/dashboard/overview/events.tsx
Processing /workspaces/nextgen/src/components/dashboard/overview/helper-widget.tsx
Processing /workspaces/nextgen/src/components/dashboard/overview/subscriptions.tsx
Processing /workspaces/nextgen/src/components/dashboard/overview/summary.tsx
Processing /workspaces/nextgen/src/components/dashboard/reports/reports-filters.tsx
Processing /workspaces/nextgen/src/components/dashboard/reports/reports-grid.tsx
Processing /workspaces/nextgen/src/components/dashboard/subscriptions/add-subscription-form.tsx
Processing /workspaces/nextgen/src/components/dashboard/subscriptions/add-subscription-schema.ts
Processing /workspaces/nextgen/src/components/dashboard/subscriptions/detail-drawer/__tests__/cost-details-card.test.tsx
Processing /workspaces/nextgen/src/components/dashboard/subscriptions/detail-drawer/__tests__/dates-card.test.tsx
Processing /workspaces/nextgen/src/components/dashboard/subscriptions/detail-drawer/__tests__/edit-section.test.tsx
Processing /workspaces/nextgen/src/components/dashboard/subscriptions/detail-drawer/__tests__/header.test.tsx
Processing /workspaces/nextgen/src/components/dashboard/subscriptions/detail-drawer/__tests__/license-info-card.test.tsx
Processing /workspaces/nextgen/src/components/dashboard/subscriptions/detail-drawer/__tests__/subscription-detail-drawer.test.tsx
Processing /workspaces/nextgen/src/components/dashboard/subscriptions/detail-drawer/__tests__/subscription-details-card.test.tsx
Processing /workspaces/nextgen/src/components/dashboard/subscriptions/detail-drawer/associated-users-drawer.tsx
Processing /workspaces/nextgen/src/components/dashboard/subscriptions/detail-drawer/cost-details-card.tsx
Processing /workspaces/nextgen/src/components/dashboard/subscriptions/detail-drawer/dates-card.tsx
Processing /workspaces/nextgen/src/components/dashboard/subscriptions/detail-drawer/edit-section.tsx
Processing /workspaces/nextgen/src/components/dashboard/subscriptions/detail-drawer/edit-subscription-schema.ts
Processing /workspaces/nextgen/src/components/dashboard/subscriptions/detail-drawer/header.tsx
Processing /workspaces/nextgen/src/components/dashboard/subscriptions/detail-drawer/license-info-card.tsx
Processing /workspaces/nextgen/src/components/dashboard/subscriptions/detail-drawer/subscription-details-card.tsx
Processing /workspaces/nextgen/src/components/dashboard/subscriptions/form-sections/basic-information.tsx
Processing /workspaces/nextgen/src/components/dashboard/subscriptions/form-sections/billing-terms.tsx
Processing /workspaces/nextgen/src/components/dashboard/subscriptions/form-sections/important-dates.tsx
Processing /workspaces/nextgen/src/components/dashboard/subscriptions/form-sections/license-details.tsx
Processing /workspaces/nextgen/src/components/dashboard/subscriptions/queries.ts
Processing /workspaces/nextgen/src/components/dashboard/subscriptions/subscription-detail-drawer.tsx
```

### Version and model info

Aider: v0.63.2

**Comment by paul-gauthier** [comment]
Thanks for trying aider and filing this issue.

Does it always get stuck at `Processing /workspaces/nextgen/src/components/dashboard/subscriptions/subscription-detail-drawer.tsx`?

If so, try deleting that file or adding it to .aiderignore and see if that helps. 

Sometimes tree sitter hangs/crashes on certain files.

**Comment by bbatchelder** [comment]
I ran into this issue as well.  It was getting stuck on a particular .tsx file.  So I deleted the file and /map completed.  Then I added the file back, used the formatting tool built into VSCode to re-format the file, and then tried /map-refresh which succeeded.

I don't normally run with --verbose, so this appeared to me at first to just be aider locking up.  Ctrl-C had no effect.  Had to kill the process.  Thankfully I found this thread.
